index.js

```diff
@@ -27,23 +73,64 @@ app.get("/image/:id", async function (req, res) {
   downloadStream.pipe(res);
 });
 
-app.post("/image/upload", upload.single("file"), async (req, res) => {
+app.post("/file/upload", upload.single("file"), async (req, res) => {
   const { file } = req;
-  // add mimetype of file and hash to metadata for every file being stored
-  // before uploading check whether hash exists
-  // hash is of multer file buffer
-  if (Buffer.from(file.path))
-    if (!file.mimetype.startsWith("image/"))
-      return res.send(415, "only image file is supported!");
+  //make video streammable before uploading in db
+  const hash = await getFileHash(file, "sha1");
+  const db = getDbClient();
+  const files = db.collection("asdad.files");
+
+  const count = await files.countDocuments({
+    metadata: { hash },
+  });
+
+  if (count) {
+    logger.info("file alreadty exists in db");
+    const fileId = await files.findOne({ metadata: { hash } }).then((doc) => {
+      return doc._id;
+    });
+    return res.send(
+      409,
+      "File already exists " + "http://localhost:" + port + "/image/" + fileId
+    );
+  }
+
+  file.fileHash = hash;
 
   const id = await insertToDb(file, file.originalname, "asdad");
-  logger.info("file " + file.originalname + " uploaded successfully!");
-  return res.send(
-    "uploaded file at :" + "http://localhost:" + port + "/image/" + id
+  logger.info(
```

Comment:
Log start time, end time and duration of request

Reply:
logged duration of req